cmd/kube-proxy/app/server.go

```diff
@@ -469,26 +470,25 @@ func NewProxyCommand() *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use: "kube-proxy",
-		Long: `The Kubernetes network proxy runs on each node. This
-reflects services as defined in the Kubernetes API on each node and can do simple
-TCP, UDP, and SCTP stream forwarding or round robin TCP, UDP, and SCTP forwarding across a set of backends.
-Service cluster IPs and ports are currently found through Docker-links-compatible
-environment variables specifying ports opened by the service proxy. There is an optional
-addon that provides cluster DNS for these cluster IPs. The user must create a service
-with the apiserver API to configure the proxy.`,
+		Long: `The Kubernetes network proxy generally should run on each node of a cluster which needs to access Service IPs. It
+reflects services (as defined in the Kubernetes API) on each node and can do simple
+TCP, UDP, and SCTP stream forwarding or round robin TCP, UDP, and SCTP forwarding to backend IP addresses, or "endpoints".
+For DNS, consult your node's DNS implementation, which usually is aware of the "kube-dns" internal service (which maps services to virtual IP addresses, which are routed via kube-proxy).
+If pods/nodes don't have access to dynamic DNS, then service IPs are redundantly defined in environment variables available in the "docker links" format (SERVICE_PORT_PROTOCOL, i.e. DB_PORT_5432_TCP) for pre-existing services.
+These variables are are injected into a pod on startup.`,
 		Run: func(cmd *cobra.Command, args []string) {
 			verflag.PrintAndExitIfRequested()
 			cliflag.PrintFlags(cmd.Flags())
 
 			if err := initForOS(opts.WindowsService); err != nil {
-				klog.Fatalf("failed OS init: %v", err)
+				klog.Fatalf("failed OS init %v, config %v", err, opts)
```

Comment:
What does this look like? I suspect `opts` will need `%#v` rather than `%v` if you want the output to actually be useful